src/compnents/PopupSignUp/PopupSignup.js

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useFormWithValidation } from '../../hooks/useFormWithValidation';
 import PopupWithForm from '../PopupWithForm/PopupWithForm';
 const PopupSignup = (props) => {
-  const { isOpen, onClose, switchPopups, onSubmit } = props;
+  const { isOpen, onClose, switchPopups, onSubmit, name } = props;
+  const { values, handleChange, errors, isValid, resetForm} = useFormWithValidation();
+  
+  // const handleChange = (event) => {
+  //   setValues({ ...values, [event.target.name]: event.target.value });
+  // };
+
+    // const handleSubmit = (event) => {
+    //   event.preventDefault();
+    //   if (isValid) {
+    //     onSubmit(values);
+    //     resetForm();
+    //   }
+    // };
+
+      // const [values, setValues] = useState({});
+
+
+      const handleSubmit = (event) => {
+        event.preventDefault();
+        if (isValid) {
+          onSubmit({
+            email: values.email,
+            password: values.password,
+            name: values.name,
+          });
+          resetForm();
```

Comment:
`        resetForm();
`

You need to **reset** the form only when you **open** it not to do that immediately after `submit` because when there is an error the popup will be opened and the user can try to send the data again to the server without having to print the data into the inputs once more



```js
    React.useEffect(() => {
        resetForm();
    }, [isOpen]);

```